ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -340,60 +347,165 @@ azure_object_size(
     return 0;
 }
 
+// File open for the Azure storage source.
 static int
 azure_file_open(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name,
   WT_FS_OPEN_FILE_TYPE file_type, uint32_t flags, WT_FILE_HANDLE **file_handlep)
 {
-    WT_UNUSED(file_system);
+    azure_file_system *azure_fs = reinterpret_cast<azure_file_system *>(file_system);
+
+    // Azure only supports opening the file in read only mode.
+    if ((flags & WT_FS_OPEN_READONLY) == 0 || (flags & WT_FS_OPEN_CREATE) != 0) {
+        std::cerr << "azure_file_open: read-only access required." << std::endl;
+        return EINVAL;
+    }
+
+    // Only data files should be opened; although this constraint can be relaxed in future.
+    if (file_type != WT_FS_OPEN_FILE_TYPE_DATA && file_type != WT_FS_OPEN_FILE_TYPE_REGULAR) {
+        std::cerr << "azure_file_open: only data file and regular types supported." << std::endl;
+        return EINVAL;
+    }
+
+    // Check if object exists.
+    bool exists;
+    int ret;
+    size_t size = 0;
+    if ((ret = azure_fs->azure_conn->object_exists(std::string(name), exists, size)) != 0) {
+        std::cerr << "azure_file_open: object_exists request to Azure failed." << std::endl;
+        return ret;
+    }
+    if (exists == false) {
+        std::cerr << "azure_file_open: no such file." << std::endl;
+        return EINVAL;
+    }
+
+    // Check if there is already an existing file handle open.
+    auto fh_iterator = std::find_if(azure_fs->azure_fh.begin(), azure_fs->azure_fh.end(),
+      [name](azure_file_handle *fh) { return strcmp(name, fh->name.c_str()) == 0; });
```

Comment:
np: I would use fh->name.compare(name) instead

Reply:
ty for the suggestion, fixed :)